File: 2019/Day 2/Task 2-1.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

using namespace std;

int main ()
{
	ifstream fin;
	fin.open("Task 2-1.txt");
	vector < int > data;
	string x;
	int pos = 0;
	if( fin.is_open() )
	{
		while( getline (fin, x, ','))
		{
			data.push_back( stoi(x) );
		}
		fin.close();
	}
	else
		cout << "Can't open the file" << endl;
		
	data[1] = 12;
	data[2] = 2;
	while( data[pos] != 99 )
	{
		switch( data[pos] ) {
			case 1 : data[data[pos + 3]] = data[data[pos + 1]] + data[data[pos + 2]];
					 break;
			case 2 : data[data[pos + 3]] = data[data[pos + 1]] * data[data[pos + 2]];
					 break;
			default : cout << "There has been a wrong input";
			}
		pos += 4;
	}
	cout << data[ 0 ] << endl;
}

File: 2019/Day 2/Task 2-1.py
def load_code():
    with open("Task 2-1.txt") as src:
        return list(map(int, src.read().split(",")))


def execute(program, noun, verb):
    pos = 0
    program[1] = noun
    program[2] = verb
    while pos < len(program):
        opcode = program[pos]
        if opcode == 1 or opcode == 2:
            op1 = program[pos + 1]
            op2 = program[pos + 2]
            dest = program[pos + 3]
            program[dest] = program[op1] + program[op2] if opcode == 1 else program[op1] * program[op2]
            pos += 4
        elif opcode == 99:
            break
        else:
            print("There has been a wrong input")
            break
    return program[0]


intcode = load_code()
print(execute(intcode, 12, 2))